### ISSUE TITLE
No declaration files in output dir

### ISSUE THREAD

**Issue by Delagen** [s-panferov/awesome-typescript-loader#317]
Using latest 3.0 beta 17

**Comment by Phmager** [comment]
Same here.

        loaders: [{
            test: /\.ts$/, 
            loaders: ['awesome-typescript-loader'],
            query: {
                useCache: false
            }
        }]

tsconfig:
{
	"compilerOptions": {
		"experimentalDecorators": true,
		"module": "commonjs",
		"target": "es5",
		"noImplicitAny": true,
		"suppressImplicitAnyIndexErrors": true,
		"moduleResolution": "node",
		"emitDecoratorMetadata": true,
		"sourceMap": true,
		"declaration": true,
		"outDir": "dist",
		"removeComments": false,
		"lib": [
			"es6",
			"dom"
		],
		"typeRoots": [
			"node_modules/@types"
		],
		"types": [
			"node"
		]
	},
	"files": [
		"src/main.ts"
	],
	"exclude": [
		"node_modules",
		"dist"
	]
}

The resulting js file is correct. Only .d.ts is missing.

**Comment by stringparts** [comment]
Any update on this?

**Comment by chiefmc** [comment]
subscribed

**Comment by loxy** [comment]
I'm facing the same problem

**Comment by stoffeastrom** [comment]
Why was this closed. Just upgraded to latest and have the same issue except the `d.ts` files is generated in the `src` folder...

**Comment by gotenxds** [comment]
Same here....

**Comment by hokkaido** [comment]
Yeah @s-panferov what's the reason for closing the issue? Worked before with 2.2.4, doesn't work anymore with 3.0.8

**Comment by mblandfo** [comment]
I think this was fixed in 3.0.3. That's the version I was on and definitions are created correctly for me. I just upgraded to 3.0.8 and it's still creating definitions correctly.

```
{ test: /\.ts$/, loader: 'awesome-typescript-loader' },


"devDependencies": {
    "awesome-typescript-loader": "3.0.8",
    ...
    "tslint": "4.3.1",
    "typescript": "2.1.5",
    "webpack": "2.2.0",
    "webpack-dev-server": "2.2.0"
  }
```

**Comment by mblandfo** [comment]
Ohhh! It's only working for me because I have this in my webpack.config.js under plugins:

```
       new DeclarationBundlerPlugin({
            moduleName: 'MyModule',
            out: './MyModule.d.ts'
        })
```
